docs/en/serving/api_server_tools.md

```diff
@@ -0,0 +1,152 @@
+# Tools
+
+## Single Round Invocation
+
+Currently, LMDeploy supports tools only for InternLM2 or InternLM2.5 models. Please start the service of models before running the following example.
+Make sure to pass skip_special_tokens as False.
+
+```python
+from openai import OpenAI
+
+tools = [
+  {
+    "type": "function",
+    "function": {
+      "name": "get_current_weather",
+      "description": "Get the current weather in a given location",
+      "parameters": {
+        "type": "object",
+        "properties": {
+          "location": {
+            "type": "string",
+            "description": "The city and state, e.g. San Francisco, CA",
+          },
+          "unit": {"type": "string", "enum": ["celsius", "fahrenheit"]},
+        },
+        "required": ["location"],
+      },
+    }
+  }
+]
+messages = [{"role": "user", "content": "What's the weather like in Boston today?"}]
+
+client = OpenAI(api_key='YOUR_API_KEY',base_url='http://0.0.0.0:23333/v1')
+model_name = client.models.list().data[0].id
+response = client.chat.completions.create(
+    model=model_name,
+    messages=messages,
+    temperature=0.8,
+    top_p=0.8,
+    stream=False,
+    tools=tools,
+    extra_body={'skip_special_tokens': False})
```

Comment:
Can we hide the `extra_body` in the implementation instead of the API?
I don't find it in the openai API spec.

Reply:
extra_body is common when calling openai api. There are documentations in vllm too. https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html#extra-parameters